CMakeLists.txt

```diff
@@ -1,187 +1,311 @@
-  endif()
-  if (SINGULARITY_USE_CUDA)
-    message(WARNING "EOSPAC not supported with cuda. You may encounter errors.")
-  endif()
+if(SINGULARITY_BUILD_SESAME2SPINER)
+  add_subdirectory(sesame2spiner)
 endif()
 
-#------------------------------------------------------------------------------#
-# Process some options
-#------------------------------------------------------------------------------#
-
-if (SINGULARITY_SUBMODULE_MODE)
-  set(BUILD_TESTING OFF CACHE BOOL "" FORCE)
-  set(CMAKE_FIND_USE_PACKAGE_REGISTRY OFF CACHE BOOL "" FORCE)
-  set(CMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY OFF CACHE BOOL "" FORCE)
-  set(SINGULARITY_BETTER_DEBUG_FLAGS OFF CACHE BOOL "" FORCE)
-  set(SINGULARITY_HIDE_MORE_WARNINGS ON CACHE BOOL "" FORCE)
+# defines
+if(SINGULARITY_USE_SINGLE_LOGS)
+  target_compile_definitions(singularity-eos PUBLIC SINGULARITY_USE_SINGLE_LOGS)
 endif()
 
-# side-projects
-if(SINGULARITY_USE_HDF5)
-  if(SINGULARITY_BUILD_STELLARCOLLAPSE2SPINER)
-    list(APPEND _subdirs2add stellarcollapse2spiner)
+if(SINGULARITY_TEST_SESAME)
+  target_compile_definitions(singularity-eos PRIVATE SINGULARITY_TEST_SESAME)
+endif()
+if(SINGULARITY_BUILD_CLOSURE)
+  target_compile_definitions(singularity-eos PRIVATE SINGULARITY_BUILD_CLOSURE)
+endif()
+
+# ------------------------------------------------------------------------------#
+# Handle dependencies
+# ------------------------------------------------------------------------------#
+
+# There are two modes for building: 1.) Submodule mode this mode is for projects
+# where singularity-eos, and it's dependencies, are present in the source tree.
+# This mode is only appropriate for projects that have been designed around
+# using git submodules for dependency management. Submodule mode disables all
+# export/install steps. 2.) Standalone mode this mode will build and install
+# singularity-eos as as a complete software package for the intended platform.
+# In standalone mode, all dependencies are expected to be found with
+# `find_package`, and an error will be produced if the packages required for
+# building are not located outside of the source directory, except for explicit
+# cases.
+
+if(SINGULARITY_USE_SPINER_WITH_HDF5)
+  singularity_enable_hdf5(singularity-eos)
+endif()
+
+if(SINGULARITY_USE_EOSPAC)
+  # NB This will add the `eospac-wrapper` directory.
+  singularity_enable_eospac(singularity-eos)
+endif()
+
+if(SINGULARITY_SUBMODULE_MODE)
+  # add all submodules
+  message(STATUS "singularity-eos configuring in submodule mode.")
+  singularity_import_mpark_variant()
+  singularity_import_ports_of_call()
+  if(SINGULARITY_USE_SPINER)
+    singularity_import_spiner()
   endif()
-  if(SINGULARITY_USE_EOSPAC AND SINGULARITY_BUILD_SESAME2SPINER)
-    list(APPEND _subdirs2add sesame2spiner)
+  if(SINGULARITY_USE_KOKKOS)
+    singularity_import_kokkos()
+    if(SINGULARITY_USE_KOKKOSKERNELS)
+      singularity_import_kokkoskernels()
+    endif()
+  else()
+    singularity_import_eigen()
```

Comment:
I think the linear algebra libraries should be hidden behind a `BUILD_CLOSURE` option.

Reply:
Updated; `SINGULARITY_BUILD_CLOSURE` is now a direct option; `SINGULARITY_USE_KOKKOSKERNELS` now depends on it, and some checks are done to enable a new cache flag `SINGULARITY_USE_EIGEN`.